src/common/urls.ts

```diff
@@ -0,0 +1,50 @@
+import ShareImageUrlJSON from 'assets/data/share_images_url.json';
+import { assert } from 'common/utils';
+
+/**
+ * We append a short unique string corresponding to the currently published
+ * sharing preview images to the URL so that if it gets shared, it's unique
+ * enough that Facebook/Twitter/whoever will fetch a recent version of the page
+ * and the up-to-date sharing image.
+ */
+const SHARING_ID_QUERY_PARAM = 's';
+const SHARING_ID = sharingId();
+const SHARING_ID_QUERYSTRING = `?${SHARING_ID_QUERY_PARAM}=${SHARING_ID}`;
+
+/**
+ * Adds unique sharing querystring to the URL.
+ *
+ * TODO(michael): Remove this and instead have dedicated functions for
+ * generating URLs to the home page / location pages, etc. and use those
+ * everywhere rather than building up URLs piecemeal from strings.
```

Comment:
FWIW I was originally planning to tackle this (hence creating this `urls.ts` helpers file), but I decided it was too much churn to put into this PR.  So I decided to punt for now.